src/unix/bsd/freebsdlike/freebsd/mod.rs

```diff
@@ -1355,6 +1357,66 @@ s! {
         pub strchange_instrms: u16,
         pub strchange_outstrms: u16,
     }
+
+    pub struct filedesc {
+        pub fd_files: *mut fdescenttbl,
+        pub fd_map: *mut c_void,
+        pub fd_freefile: c_int,
+        pub fd_refcnt: c_int,
+        pub fd_holdcnt: c_int,
+        pub fd_sx: sx,
```

Comment:
You should make this field private and opaque.  It isn't used by userspace.

Reply:
I made them private, but wdym by opaque? To get the size and alignment right, we still need to define the inner types, or do you mean something else?

Reply:
You should be able to get the size and alignment correct with a suitably constructed array of bytes, longs, etc.  But if that's too much work then it's ok to define the real structs instead.

Reply:
I find it preferrable to have definitions matching C as it makes it easier for us to update it if changed but really as you prefer.

Reply:
For 1.0 we are going to make all opaque / padding fields `MaybeUninit`, that could be an option here